### DISCUSSION TITLE
How to send image from base64 ?

### DISCUSSION THREAD

**Discussion by frmdeveloper** [tulir/whatsmeow#218] [WhatsApp protocol Q&A]
I have images, but in base64 format.
I don't know how to upload to whatsmeow

**Comment by mamur-rezeki** [comment]
` whatsmeow ` working with ` []byte ` to be uploaded directly to the `  client `.

You can decode it into buffer ` []byte ` type then upload it as what the filetype is.

You can read on #204 issue about How to upload and filling the fields if needed.

Actually that is available in [mdtest](https://github.com/tulir/whatsmeow/tree/main/mdtest) 
https://github.com/tulir/whatsmeow/blob/aaaedc8d029ab674f438cc9374f9ad9eee012f25/mdtest/main.go#L506-L535

**Reply by frmdeveloper** [reply]
By the way, mine has an error. cli.Upload returns only url and direct_path.

MediaKey, FileEncSHA256, FileSHA256 missing

**Reply by mamur-rezeki** [reply]
Upload always provided by complete needed field.

Have you checked the error condition?
